src/widget/wwaveformviewer.cpp

```diff
@@ -91,6 +98,12 @@ void WWaveformViewer::mousePressEvent(QMouseEvent* event) {
                         m_pCueMenuPopup->size());
                 m_pCueMenuPopup->popup(cueMenuTopLeft);
             }
+
+            if (m_hoveredBeat.isEnabled()) {
+                m_pBeatMenu->update();
+                m_pBeatMenu->popup(event->globalPos());
+            }
```

Comment:
Hmm... how should we handle this situation?
![two-menus](https://user-images.githubusercontent.com/9455094/84335650-f12ebd80-ab5a-11ea-854e-76e0935634b0.png)

Reply:
I have two ideas how this could be handled:
1. Show WBeatMenu and add QAction "Edit cue" that shows WCueMenuPopup.
2. Add actions to WCueMenuPopup for each of WBeatMenu's actions.

I'm leaning towards 1. What do others think? @Holzhaus, @ronso0?

Reply:
I kinda had 1 in mind earlier since I was aware of this situation. I was just thinking it wouldn't be appropriate to move the cue menu as an action in BeatMenu since a cue can also be set at a position where there is no beat (if quantization is disabled).

Reply:
Right, this only only be for the special case where a cue and beat are at the same point (which is a common scenario with quantize). For cues that are not at beats, simply show WCueMenuPopup by itself like is done currently.

Reply:
1. requires an additional click, but I think it makes sense here.